packages/fx-core/src/component/deploy/azureStorageDeployDriver.ts

```diff
@@ -0,0 +1,183 @@
+
+import { AzureDeployDriver } from "./azureDeployDriver";
+import { AzureResourceInfo, DeployStepArgs, DriverContext } from "../interface/buildAndDeployArgs";
+import { TokenCredentialsBase } from "@azure/ms-rest-nodeauth";
+import {
+  BlobDeleteResponse,
+  BlobItem,
+  BlobServiceClient,
+  BlobUploadCommonResponse,
+  BlockBlobParallelUploadOptions,
+  ContainerClient,
+} from "@azure/storage-blob";
+import {
+  StorageAccounts,
+  StorageManagementClient,
+  StorageManagementModels,
+} from "@azure/arm-storage";
+import { DeployConstant } from "../constant/deployConstant";
+import { DeployExternalApiCallError } from "../error/deployError";
+import { forEachFileAndDir } from "../utils/fileOperation";
+import * as fs from "fs-extra";
+import path from "path";
+import * as mime from "mime";
+import { LogProvider } from "@microsoft/teamsfx-api";
+import { Service } from "typedi";
+import { StepDriver } from "../interface/stepDriver";
+
+@Service("deploy/azureStorage")
+export class AzureStorageDeployDriver implements StepDriver {
+  async run(args: unknown, context: DriverContext): Promise<Map<string, string>> {
+    const impl = new AzureStorageDeployDriverImpl(args, context);
+    return await impl.run();
+  }
+}
+
+/**
+ * deploy to Azure Storage
+ */
+export class AzureStorageDeployDriverImpl extends AzureDeployDriver {
+  pattern =
+    /\/subscriptions\/([^\/]*)\/resourceGroups\/([^\/]*)\/providers\/Microsoft.Storage\/storageAccounts\/([^\/]*)/i;
+
+  async azureDeploy(
+    args: DeployStepArgs,
+    azureResource: AzureResourceInfo,
+    azureCredential: TokenCredentialsBase
+  ): Promise<void> {
+    const containerClient = await AzureStorageDeployDriverImpl.createContainerClient(
+      azureResource,
+      azureCredential
+    );
+    // delete all existing blobs
+    await AzureStorageDeployDriverImpl.deleteAllBlobs(
+      containerClient,
+      azureResource.instanceId,
+      this.context.logProvider
+    );
+    // upload all to storage
+    const ig = await this.handleIgnore(args, this.context);
+    const sourceFolder = args.dist;
+    const tasks: Promise<BlobUploadCommonResponse>[] = [];
+    await forEachFileAndDir(
+      sourceFolder,
+      (filePath: string, stats: fs.Stats) => {
+        const destFilePath: string = path.relative(sourceFolder, filePath);
+        if (!destFilePath || stats.isDirectory()) {
+          return;
+        }
+        const options: BlockBlobParallelUploadOptions = {
+          blobHTTPHeaders: {
+            blobContentType: mime.getType(filePath) || undefined,
+          },
+        };
+        const client = containerClient.getBlockBlobClient(destFilePath);
+        tasks.push(client.uploadFile(filePath, options));
+      },
+      (itemPath: string) => {
+        return ig.test(path.relative(sourceFolder, itemPath)).unignored;
+      }
+    );
+    const responses = await Promise.all(tasks);
+    const errorResponse = responses.find((res) => res.errorCode !== undefined);
```

Comment:
In TypeScript, it is recommend to use !res.errorCode instead of directly comparing with undefined, since there are other "empty value" besides undefined.

Reply:
Ok, changed to !res.errorCode